app/code/Magento/Analytics/ReportXml/QueryFactory.php

```diff
@@ -45,36 +49,44 @@ class QueryFactory
      */
     private $selectHydrator;
 
+    /**
+     * @var Json|null
+     */
+    private $jsonSerializer;
+
     /**
      * QueryFactory constructor.
      *
-     * @param CacheInterface $queryCache
-     * @param SelectHydrator $selectHydrator
+     * @param CacheInterface         $queryCache
+     * @param SelectHydrator         $selectHydrator
      * @param ObjectManagerInterface $objectManager
-     * @param SelectBuilderFactory $selectBuilderFactory
-     * @param Config $config
-     * @param array $assemblers
+     * @param SelectBuilderFactory   $selectBuilderFactory
+     * @param Config                 $config
+     * @param array                  $assemblers
+     * @param Json|null              $jsonSerializer
      */
     public function __construct(
         CacheInterface $queryCache,
         SelectHydrator $selectHydrator,
         ObjectManagerInterface $objectManager,
         SelectBuilderFactory $selectBuilderFactory,
         Config $config,
-        array $assemblers
+        array $assemblers,
+        Json $jsonSerializer = null
     ) {
         $this->config = $config;
         $this->selectBuilderFactory = $selectBuilderFactory;
         $this->assemblers = $assemblers;
         $this->queryCache = $queryCache;
         $this->objectManager = $objectManager;
         $this->selectHydrator = $selectHydrator;
+        $this->jsonSerializer = $jsonSerializer ?: ObjectManager::getInstance()->get(Json::class);
```

Comment:
Please remove fallback to object manager and make this parameter not optional. There it’s not needed as this class in not marked as `@api`